lib/magma.gi

```diff
@@ -1209,6 +1209,9 @@ InstallMethod( AsMagma,
     D := AsSSortedList( D );
     L := ShallowCopy( D );
     M := Submagma( MagmaByGenerators( D ), [] );
+    if IsGroup(M) then
+      M:=Submagma(MagmaByGenerators(D),[One(ElementsFamily(FamilyObj(D)))]);
+    fi;
```

Comment:
If I were to look at this code in a year, I'd be completely confused as to what it's about. Even if I spend the extra effort to use `git annotate` to find the commit message and look at it, I don't think I'd be much wiser: "... replacing the empty submagma with the one-submagma, if it turns out a group" -- Huh? How can the submagma turn out to be a group, if it is empty? Isn't it a bug that it's marked as a group in the first place? Or is the bug that it's marked as a group"?

In fact, reading up on #2400 again, I am not convinced this change is correct. Rather, it looks like a workaround for a deeper issue. As such, I don't feel comfortable merging it, until somebody explains why it is correct after all (and ideally that explanation then should be part of a comment before this `if/fi` block).

Reply:
Yes, it is a workaround (and I've added a comment to this effect). The problem is that this bug arises in test suites (in examples of considering groups as magmas) and has the potential of holding up other commits, while it is completely unclear to me how one would fix this.